pvlib/pvarray.py

```diff
@@ -0,0 +1,220 @@
+"""
+This module contains implementations of PV module and array electrical models.
+
+These models are used to predict the electrical behavior of pv modules
+or collections of pv modules (arrays).  The primary inputs are
+effective irradiance and operating temperature and the outputs may range from
+power or efficiency at the maximum power point to complete IV curves.
+Supporting functions and parameter fitting functions may also be found here.
+"""
+
+import numpy as np
+from scipy.optimize import curve_fit
+from scipy.special import exp10
+
+
+def pvefficiency_adr(effective_irradiance, temp_cell,
+                     k_a, k_d, tc_d, k_rs, k_rsh):
+    '''
+    Calculate PV module efficiency using the ADR model.
+
+    The efficiency varies with irradiance and operating temperature
+    and is determined by 5 model parameters as described in [1]_.
+
+    Parameters
+    ----------
+    effective_irradiance : numeric, non-negative
+        The effective irradiance incident on the PV module. [W/m²]
```

Comment:
I am not in favor of the superscript/special characters. I think we discussed this in the past but I can't locate it. If my memory is not correct and we agreed this was OK, please disregard my comment.

```suggestion
        The effective irradiance incident on the PV module. [W/m^2]
```

Reply:
Not worth searching my memory.  I know which one I think looks better but I'll change it.